rezz.lic

```diff
@@ -1,118 +1,131 @@
-    find_soul(person, mana)
-
-    # snap cast soul bonding
-    cast_soulbonding(person)
-
-    # check for silver again. recast rejuv (or infuse again) if not
-    rejuv(person)
-
-    # gesture
-    bput("gest #{person}", 'Roundtime')
+  
+  def check_favor
+    Flags.add('no-favor', /(#{@person}.*) has no favor with (.*) god/)
+    bput("perc #{@person}", /#{@person} has (.*) favors with (his|her) god/)
+    Flags['no-favor'] ? begin echo "*** #{@person} has no favors! ***"; exit end : nil
   end
-
-  def find_soul(person, infuse_amt)
-    loop do
-      pause 5 while mana <= 40
-      harness_mana([infuse_amt]) unless skip_harness?
-      case bput("infuse rezz #{infuse_amt}", /You clearly sense the spirit of #{person} in the Void/i, 'You sense a spirit nearby, but you are unable to make it out clearly.', 'Your sense of sight has not fully transitioned to the spirit world', "You don't have enough harnessed mana to infuse that much.")
-      when 'You sense a spirit nearby, but you are unable to make it out clearly.'
-        next
-      when /You clearly sense the spirit of #{person} in the Void/i
-        break
-      when 'Your sense of sight has not fully transitioned to the spirit world'
-        pause 2
-        next
-      when "You don't have enough harnessed mana to infuse that much."
-        next
-      else
-        next
-      end
+  
+  def rejuv(quick = false)
+    quick ? begin @rejuv['cambrinth'] = []; @rejuv['prep_time'] = 5; @rejuv['mana'] = 5 end : nil
+    Flags.add('wrong-name', "You can't cast that at yourself")
+    Flags.add('silver', 'A thin silver nimbus surrounds')
+    Flags.add('alive', 'Nothing happens')
+    until Flags['silver'] || Flags['alive'] || Flags['wrong-name']
+	  pause 1 while DRStats.mana < 50
+      cast_spell(@rejuv, @settings)
+  	  pause 0.5
     end
+    Flags['alive'] ? begin echo "*** #{@person} is already alive, dumbass! ***"; exit end : nil
+    Flags['wrong-name'] ? begin echo "*** Name spelled incorrectly. ***"; exit end : nil
+    Flags.reset('rejuv-silver')
   end
-
-  def is_silver?(person)
-    /As you concentrate, a thin (.*) nimbus/ =~ bput("perc #{person}", 'As you concentrate, a thin (.*) nimbus')
-    color = Regexp.last_match(1)
-    echo "Memories are #{color}"
-    waitrt?
-    color.eql? 'silver'
-  end
-
-  def rejuv(person)
-    return if is_silver?(person)
-    data = get_data('spells').spell_data['Rejuvenation']
-    data['cast'] = "cast #{person}"
-    if @rejuv_mana.zero?
-      data = check_discern(data, @settings)
-      @rejuv_mana = data['mana']
+    
+  def cast_rezz
+    Flags.add('rezz-ready', 'You prepare to look for the lost spirits of the dead')
+    Flags.add('rezz-not-prepped', 'This pattern may only be cast with full preparation')
+    cast_spell(@rezz, @settings)
+    if Flags['rezz-not-prepped']
+      while Flags['rezz-not-prepped'] 
+	    Flags.reset('rezz-not-prepped')
+        pause 1
+	    bput('cast', get_data('spells').cast_messages, )
```

Comment:
There's an extra comma at the end of that statement, before the ).  I will take that out.

Reply:
I made all of the different methods because that's how the original script was setup for the most part.

I'll look over the suggested changes.  Thanks for the feedback!

Reply:
I think the program adage is, "refactor, reduce", or I could be pulling that out of my ass.